### PR TITLE
Added some HTML parsing to diff command

### PR DESCRIPTION
**Pull request by Friskygote**
Note, please do not merge this pull request yet. It lacks a package in package.json and some really good cleaning, potentially optimization. Also, removing debug statements I've made would be a great idea.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1174,29 +1175,119 @@ function cmd_diffsend(lang, msg, args, wiki, reaction, spoiler) {
...
-					}
-					else {
-						var embed = {};
-						var text = '<' + pagelink + '>\n\n' + editor.join(' ') + '\n' + timestamp.join(' ') + '\n' + size.join(' ') + '\n' + comment.join(' ') + ( tags ? '\n' + tags.join(' ').replace( tagregex, '$2' ) : '' );
+					request( {
+						uri: wiki + 'api.php?action=compare&format=json&fromrev='+ oldid + '&torev='+ diff,
+						json: true
+					}, function( cperror, cpresponse, cpbody ) {
+						if ( cpbody && cpbody.warnings ) log_warn(cpbody.warnings);
+						if ( error || !cpresponse || cpresponse.statusCode !== 200 || !cpbody || !cpbody.compare ) {
```
Comment (Friskygote) [inline main.js]: A lot of this error handling is directly copied from above

File: main.js [hunk 2]
```diff
@@ -1174,29 +1175,119 @@ function cmd_diffsend(lang, msg, args, wiki, reaction, spoiler) {
...
+								console.log( '- This wiki doesn\'t exist! ' + ( cperror ? cperror.message : ( cpbody ? ( cpbody.error ? cpbody.error.info : '' ) : '' ) ) );
+								msg.reactEmoji('nowiki');
+							}
+							else {
+								console.log( '- ' + ( cpresponse ? cpresponse.statusCode + ': ' : '' ) + 'Error while getting the search results' + ( cperror ? ': ' + cperror : ( cpbody ? ( cpbody.error ? ': ' + cpbody.error.info : '.' ) : '.' ) ) );
+								msg.sendChannelError( spoiler + '<' + wiki.toLink() + 'Special:Diff/' + ( args[1] ? args[1] + '/' : '' ) + args[0] + '>' + spoiler );
+							}
+						} else {
+							var content_diff = cpbody.compare["*"];
+							var current_tag = "";
```
Comment (Friskygote) [inline main.js]: I'm not sure if assigning so many variables is optimal, however when I've made my own version of this htmlparser it was the only way I knew

File: main.js [hunk 3]
```diff
@@ -1174,29 +1175,119 @@ function cmd_diffsend(lang, msg, args, wiki, reaction, spoiler) {
...
+									}
+									console.log("text: " + text) // TODO Remove debug
+								},
+								onclosetag: function(tagname){
+									console.log("onclosetag name: " + tagname); // TODO Remove debug
+									if (tagname === "ins") {
+										current_tag = "afterins";
+									} else if (tagname === "del") {
+										current_tag = "afterdel";
+									} else if (tagname === "tr" && added === 2) {
```
Comment (Friskygote) [inline main.js]: This if is problematic, if the edit only removed lines without adding anything the diff in removed side won't have line breaks

**Comment by Markus-Rost** [comment]
Request should now be ready for review

**Pull request merged**